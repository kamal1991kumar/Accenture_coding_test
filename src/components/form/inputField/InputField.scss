.inputFiled {
  &__input {
    width: 100%;
    border:solid 1px #ebebeb;
    padding: 8px 10px;
    font-size: 12px;
    line-height: 15px;
    text-transform: capitalize;
    &-error {
      border-color: red;
    }
  }
  &__error {
    color: red;
    text-transform: capitalize;
    font-size: 8px;
    padding: 3px;
  }
}