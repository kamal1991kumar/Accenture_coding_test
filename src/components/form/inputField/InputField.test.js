import React from "react";
import { render, fireEvent } from "@testing-library/react";
import InputField from "./index";

const props = {
  hasError: false,
  message: "input not be empty",
  value: "text",
  handleChange: jest.fn(),
};

describe("InputFiled Component", () => {
  test("Should render on screen", () => {
    const { container } = render(<InputField {...props} />);
    expect(container).toMatchSnapshot();
  });

  test("Should render props value", () => {
    const { getByTestId } = render(<InputField {...props} />);
    expect(getByTestId("input")).toHaveValue(props.value);
  });

  test("Should render changed value", () => {
    const { getByTestId } = render(<InputField {...props} />);
    const input = getByTestId("input");
    fireEvent.change(input, {
      target: { value: "123" },
    });
    expect(input).toHaveValue("123");
  });
  test("Should render error message", () => {
    const { container, getByTestId } = render(
      <InputField {...props} hasError />
    );
    const input = getByTestId("error");
    expect(container).toMatchSnapshot();
    expect(input.textContent).toBe(props.message);
  });
});
