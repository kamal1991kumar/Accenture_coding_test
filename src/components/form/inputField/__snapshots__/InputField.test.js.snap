// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFiled Component Should render error message 1`] = `
<div>
  <div
    class="inputFiled"
  >
    <input
      class="inputFiled__input inputFiled__input-error"
      data-testid="input"
      type="text"
      value="text"
    />
    <div
      class="inputFiled__error"
      data-testid="error"
    >
      input not be empty
    </div>
  </div>
</div>
`;

exports[`InputFiled Component Should render on screen 1`] = `
<div>
  <div
    class="inputFiled"
  >
    <input
      class="inputFiled__input inputFiled__input-"
      data-testid="input"
      type="text"
      value="text"
    />
  </div>
</div>
`;
